### ISSUE TITLE
Issue with ascriptions with the two phase typechecker

### ISSUE THREAD

**Issue by aseemr** [FStarLang/FStar#1349]
@tahina-pro has reported the following example that doesn't work with the two phase type checker:

```
let test (b: bool) = if b then pos else nat

 //fails
let coerce (b: bool) (t: test b) : Tot (test false) = (t <: nat) 

//fails
let coerce (b: bool) (t: test b) : Tot (test false) = (t <: nat) <: test false

//succeeds
let coerce (b: bool) (t: test b) : Tot (test false) =
  let t' : nat = t in t'
```

**Comment by aseemr** [comment]
The issue here is that after the first phase of lax checking, we normalize the term to reduce unification solutions. The normalizer, when normalizing an ascribed term, drops the inner ascriptions, essentially leaving the term `t <: test false`, which does not go through (even in the single phase, the ascription `<: nat` is necessary). The ascription itself is needed because of the bug#606.

I discussed with @nikswamy, and we thought it's better to leave the normalizer as is (else we will end up keeping the ascriptions unnecessarily). As a workaround, following would work:

```
let test (b: bool) = if b then pos else nat

let as_nat (b:bool) (t:test b) :nat = t

let coerce (b: bool) (t: test b) : Tot (test false) = as_nat t
```
Essentially a more verbose ascription.

**Comment by aseemr** [comment]
With fix for #606, the examples above go through without any ascriptions. Closing this issue.
